sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnInvoker.java

```diff
@@ -56,11 +56,13 @@
   <RestrictionT> RestrictionT invokeGetInitialRestriction(InputT element);
 
   /** Invoke the {@link DoFn.GetRestrictionCoder} method on the bound {@link DoFn}. */
-  <RestrictionT> Coder<RestrictionT> invokeGetRestrictionCoder();
+  <RestrictionT> Coder<RestrictionT> invokeGetRestrictionCoder(CoderRegistry coderRegistry);
```

Comment:
Add a comment about when this should be invoked. Generally, we shouldn't call methods that require a `CoderRegistry` on the worker, since it isn't available there. Maybe clarify here (and on `getRestrictionCoder` that this is called during pipeline construction)?